src/components/Requirements/IncompleteSelfCheck.vue

```diff
@@ -170,10 +175,12 @@ export default defineComponent({
           // We also need to opt-out of all requirements without warnings,
           // because the user intention is clear that we only want to bind
           // the course to this specific requirement.
-          optOut: getAllEligibleRelatedRequirementIds(
+          optOut: getRelatedRequirementIdsForCourseOptOut(
             newCourse.crseId,
+            this.subReqId,
             store.state.groupedRequirementFulfillmentReport,
-            store.state.toggleableRequirementChoices
+            store.state.toggleableRequirementChoices,
+            store.state.userRequirementsMap
           ),
```

Comment:
This fixes the bug in #626.
College: ENG
Major: CS

Drag MATH 4710 from the CHEM 2080 requirement to your schedule. The CS probability requirement should also be connected.

Reply:
This works!